bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/AutoUpdateManager.java

```diff
@@ -174,27 +176,41 @@ public void receiveCommand(ItemCommandEvent commandEvent, Item item) {
     private Recommendation shouldAutoUpdate(String itemName) {
         Recommendation ret = Recommendation.REQUIRED;
 
-        Set<ChannelUID> linkedChannelUIDs = itemChannelLinkRegistry.stream().filter(link -> {
-            return link.getItemName().equals(itemName);
-        }).map(link -> {
-            return link.getLinkedUID();
-        }).collect(toSet());
+        List<ChannelUID> linkedChannelUIDs = new ArrayList<>();
+        for (ItemChannelLink link : itemChannelLinkRegistry.getAll()) {
+            if (link.getItemName().equals(itemName)) {
+                linkedChannelUIDs.add(link.getLinkedUID());
+            }
+        }
+
+        // check if there is any channel ONLINE
+        List<ChannelUID> onlineChannelUIDs = new ArrayList<>();
         for (ChannelUID channelUID : linkedChannelUIDs) {
             Thing thing = thingRegistry.get(channelUID.getThingUID());
             if (thing == null //
                     || thing.getChannel(channelUID.getId()) == null //
                     || thing.getHandler() == null //
                     || !ThingStatus.ONLINE.equals(thing.getStatus()) //
             ) {
-                if (ret == Recommendation.REQUIRED) {
-                    ret = Recommendation.REVERT;
-                }
                 continue;
             }
+            onlineChannelUIDs.add(channelUID);
+        }
+        if (!linkedChannelUIDs.isEmpty() && onlineChannelUIDs.isEmpty()) {
+            // none of the linked channels is able to process the command
```

Comment:
Just to get it right: What about things that online detection works only reliable as soon as a command has been sent to the device? Haven't you told me some time that we cannot really rely on the thing online / offline status as it can already be outdated after it has been set?